src/encord_active/lib/common/data_utils.py

```diff
@@ -146,10 +147,42 @@ def download_file(
     return destination
 
 
-def download_image(url: str) -> Image.Image:
+def url_to_file_path(url: str, project_dir: Path) -> Optional[Path]:
+    if url.startswith("/") or url.startswith("./"):
+        return Path(url)
     if url.startswith("file:"):
-        image_path = Path(unquote(urlparse(url).path))
-        return Image.open(image_path)
+        return Path(unquote(urlparse(url).path))
+    if url.startswith("relative://"):
+        relative_path = url[len("relative://") :]
+        return project_dir / Path(relative_path)
+    if url.startswith("absolute://"):
+        absolute_path = url[len("absolute:/") :]
+        return Path(absolute_path)
+    return None
+
+
+def file_path_to_url(path: Path, project_dir: Path, relative: Optional[bool] = None) -> str:
+    path = path.absolute().resolve()
+    if relative is not None and not relative:
+        return path.as_uri()
+
+    # Attempt to create relative path
+    root_path = project_dir.absolute().resolve()
+    try:
+        rel_path = path.relative_to(root_path)
+        return f"relative://{rel_path.as_posix()}"
+    except ValueError:
+        if relative is not None:
+            raise
+
+    # Fallback to strict uri
+    return path.as_uri()
+
+
+def download_image(url: str, project_dir: Path) -> Image.Image:
+    path = url_to_file_path(url, project_dir)
+    if path is not None:
```

Comment:
would the path always exist?

Reply:
no - the path could be remote (or for remote encord projects a signed url)